docs/aws/audit/elbmonitoring/rules/elb_app_tier_security_policy_remediation.mdx

```diff
@@ -1,6 +1,190 @@
+                ListenerArn=listener_arn,
+                SslPolicy=latest_ssl_policy
+            )
+            print(f"Updated SSL policy for listener {listener_arn} to {latest_ssl_policy}")
+```
+
+### Summary
+1. **Install Boto3**: Ensure Boto3 is installed in your Python environment.
+2. **Initialize Boto3 Client**: Set up the Boto3 client for ELB.
+3. **Retrieve List of Load Balancers**: Get the list of load balancers.
+4. **Update SSL Policy**: Loop through the load balancers and update their SSL policies to the latest recommended policy.
+
+This script will help you ensure that all your App-Tier ELBs are using the latest AWS security policy for SSL negotiations.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and navigate to the EC2 dashboard.
+2. In the navigation pane, under "Load Balancing", click on "Load Balancers".
+3. Select the load balancer you want to inspect. In the "Description" tab below, find the "Listeners" section.
+4. Check the "SSL Certificate" column. If the SSL certificate is not listed for the HTTPS or SSL protocols, it means that the latest AWS security policy for SSL negotiations is not being used.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the Elastic Load Balancer (ELB) resources.
+
+2. Once the AWS CLI is set up, you can list all the load balancers in your account using the following command:
+
+   ```
+   aws elbv2 describe-load-balancers
+   ```
+
+   This command will return a JSON object with details about all your load balancers.
+
+3. To check the SSL policies of each load balancer, you need to describe the listeners for each load balancer. You can do this using the following command:
+
+   ```
+   aws elbv2 describe-listeners --load-balancer-arn <load-balancer-arn>
+   ```
+
+   Replace `<load-balancer-arn>` with the ARN of your load balancer. This command will return a JSON object with details about the listeners for your load balancer.
+
+4. To specifically check the SSL policy, you need to look at the 'SslPolicy' field in the JSON object returned by the previous command. If the 'SslPolicy' field is not set to the latest security policy, then the load balancer is not using the latest SSL negotiations.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can do this by running the command `pip install boto3` in your terminal.
+
+2. Import the necessary modules and establish a session with AWS using your credentials. Here's a basic setup:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>